### PR TITLE
Eth1Data VoteCount Fix

### LINKED ISSUE #3073: Reg Sync: incorrect outstanding deposits in block body, wanted: 0, got: 16
[2019-07-25 10:39:16] ERROR regular-sync: block failed processing: could not process block: could not process block operation: could not verify operation lengths: incorrect outstanding
deposits in block body, wanted: 0, got: 16 method=github.com/prysmaticlabs/prysm/beacon-chain/sync.(*RegularSync).receiveBlock-fm

### PR DESCRIPTION
**Pull request by nisdas**
Resolves #3073

- [x] Fixes this by adding eth1Data vote to the state in block proposal.
- [x] Disable eth1Data cache

### CODE REVIEW

**Comment by rauljordan** [comment]
This is great - but I think we might need a regression test before merge @nisdas

File: beacon-chain/rpc/proposer_server.go [hunk 1]
```diff
@@ -268,6 +268,9 @@ func (ps *ProposerServer) deposits(ctx context.Context, currentVote *ethpb.Eth1D
 		return nil, fmt.Errorf("could not fetch beacon state: %v", err)
 	}
 	var eth1BlockHash [32]byte
+
+	// Add in current vote, to get accurate vote tally
+	beaconState.Eth1DataVotes = append(beaconState.Eth1DataVotes, currentVote)
```
Comment (prestonvanloon) [inline beacon-chain/rpc/proposer_server.go]: We need a regression test for this. It's too easy for this to get reverted

**Comment by prestonvanloon** [comment]
I don't think this fully resolves #3073. I'm still having the same issues with this branch changes.

**Comment by prestonvanloon** [comment]
Having difficulty to reliably reproduce the issue. I suspect it has to do with other caching implementations.

**Comment by prestonvanloon** [comment]
Please consider reverting eth1data cache changes in favor of https://github.com/prysmaticlabs/prysm/pull/3107. That PR adds a feature flag to enable / disable the cache.

File: beacon-chain/rpc/proposer_server.go [hunk 2]
```diff
@@ -332,6 +322,30 @@ func (ps *ProposerServer) deposits(ctx context.Context, currentVote *ethpb.Eth1D
 	return pendingDeposits, nil
 }
 
+// determineLatestETH1Height determines what the latest eth1Blockhash is by tallying the votes in the
+// beacon state
+func (ps *ProposerServer) determineLatestETH1Height(ctx context.Context, beaconState *pbp2p.BeaconState,
```
Comment (terencechain) [inline beacon-chain/rpc/proposer_server.go]: suggest changing the name to `latestEth1Height`, the word `determine` sounds a bit redundant there

```suggestion
func (ps *ProposerServer) latestEth1Height(ctx context.Context, beaconState *pbp2p.BeaconState,
```

File: beacon-chain/rpc/proposer_server_test.go [hunk 3]
```diff
@@ -470,6 +470,97 @@ func TestPendingDeposits_UnknownBlockNum(t *testing.T) {
...
+
+	newState, err = b.ProcessEth1DataInBlock(beaconState, blk)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !proto.Equal(newState.Eth1Data, vote) {
+		t.Errorf("eth1data in the state not of the expected kind: Got %v but wanted %v", newState.Eth1Data, vote)
+	}
+
```
Comment (terencechain) [inline beacon-chain/rpc/proposer_server_test.go]:
```suggestion
```

**Pull request merged**